pkg/sql/pgwire/conn.go

```diff
@@ -311,11 +312,25 @@ func (c *conn) serveImpl(
 	if sqlServer != nil {
 		wg.Add(1)
 		go func() {
+			defer func() {
+				if sqlServer.GetExecutorConfig().TestingKnobs.CatchPanics {
+					if r := recover(); r != nil {
+						// Catch the panic and return it to the client as an error.
+						err := pgerror.NewErrorf(pgerror.CodeCrashShutdownError, "caught fatal error: %v", r)
+						err.Detail = string(debug.Stack())
+						_ = writeErr(err, &c.msgBuilder, &c.writerState.buf)
+						_ /* n */, _ /* err */ = c.writerState.buf.WriteTo(c.conn)
+						c.stmtBuf.Close()
+						// Send a ready for query to make sure the client can react.
+						c.bufferReadyForQuery('I')
+					}
+				}
+				cancelConn()
```

Comment:
The order of the cancelConn and wg.Done were reversed previously. Does it matter that they are switched?

Reply:
Gah, fixed. My attention to detail is lacking on this change.